src/libServer/StatusServer.cpp

```diff
@@ -203,26 +203,40 @@ Json::Value StatusServer::IsTxnInMemPool(const string& tranID) {
     TxnHash tranHash(tranID);
     Json::Value _json;
 
-    switch (m_mediator.m_node->IsTxnInMemPool(tranHash)) {
-      case PoolTxnStatus::NOT_PRESENT:
-        _json["present"] = false;
-        _json["code"] = PoolTxnStatus::NOT_PRESENT;
-        return _json;
-      case PoolTxnStatus::PRESENT_NONCE_HIGH:
-        _json["present"] = true;
-        _json["code"] = PoolTxnStatus::PRESENT_NONCE_HIGH;
-        _json["info"] = "Nonce too high";
-        return _json;
-      case PoolTxnStatus::PRESENT_GAS_EXCEEDED:
-        _json["present"] = true;
-        _json["code"] = PoolTxnStatus::PRESENT_GAS_EXCEEDED;
-        _json["info"] = "Could not fit in as microblock gas limit reached";
-        return _json;
-      case PoolTxnStatus::ERROR:
-        throw JsonRpcException(RPC_INTERNAL_ERROR, "Processing transactions");
-      default:
-        throw JsonRpcException(RPC_MISC_ERROR, "Unable to process");
+    const auto& code = m_mediator.m_node->IsTxnInMemPool(tranHash);
+
+    if (!IsPoolTxnDropped(code)) {
```

Comment:
User might wonder why some results have an "info" field while the rest don't. I would prefer no "info" across the board, as long as the API is sufficiently documented for developers.